options/internal/include/bits/types.h

```diff
@@ -325,39 +325,39 @@ typedef __SIZE_TYPE__    __mlibc_size;
 /* ---------------------------------------------------------------------------- */
 
 #if defined(__cpp_static_assert)
-#	define __MLIBC_STATIC_ASSERT(c, text) static_assert(c, text)
-#elif !defined(__cplusplus)
-#	define __MLIBC_STATIC_ASSERT(c, text) _Static_assert(c, text)
+#	define __MLIBC_STATIC_ASSERT(c, text) static_assert(c, text);
```

Comment:
semicolon added because `__MLIBC_CHECK_TYPE(foo);` causes a 'redundant semi-colon' error in C89 mode

Reply:
I think this is the wrong fix. The trailing semicolon should be at the call site.